tests/app/service/test_rest.py

```diff
@@ -590,6 +727,127 @@ def test_update_service(client, notify_db, sample_service):
     assert result['data']['organisation_type'] == 'other'
 
 
+def test_update_service_with_valid_provider(
+        admin_request, notify_db, sample_service, ses_provider, current_sms_provider
+):
+    data = {
+        'email_provider_id': str(ses_provider.id),
+        'sms_provider_id': str(current_sms_provider.id),
+    }
+
+    resp = admin_request.post(
+        'service.update_service',
+        service_id=sample_service.id,
+        _data=data,
+        _expected_status=200
+    )
+    assert resp['data']['email_provider_id'] == str(ses_provider.id)
+    assert resp['data']['sms_provider_id'] == str(current_sms_provider.id)
+
+    service_db = Service.query.get(resp['data']['id'])
```

Comment:
Should we just mock the dao method and assert it's called instead of having an integration test that goes all the way to DB?